### ISSUE TITLE
How to deal with background images during DOTAv1 dataset training

### ISSUE THREAD

**Issue by yangershuai627** [ultralytics/ultralytics#20057]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

When training DOTA dataset, we usually crop them first, dividing each image into many subgraphs. However, not every subgraph in the same picture contains the target, which will lead to a lot of empty samples in the training process, and the number of empty samples has exceeded the samples containing the target. On the premise of not eliminating empty samples, the mAP50 obtained by me is 75.7, which is lower than the 78.0 of the official website. So I guess a large number of empty samples will affect the training accuracy of the model.
I want to know if the number of empty samples has far exceeded the samples containing the target, should I eliminate them or eliminate part of the empty samples, and what proportion should be between the target samples and the empty samples?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yangershuai627, thank you for reaching out and for your interest in Ultralytics 🚀!

We recommend checking out the [Docs](https://docs.ultralytics.com/) for detailed training guidance, including [custom dataset preparation](https://docs.ultralytics.com/datasets/) and [training tips](https://docs.ultralytics.com/guides/model-training-tips/), which may address your questions about dataset balance and empty samples.

If this is a ❓ Question regarding custom training, please share as much detail as possible, including:
- Dataset details (e.g., sample images or distribution)
- Training logs and hyperparameters
- Specific steps you've already taken

This will help us provide more tailored advice.

If you believe this is a 🐛 Bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate. 

## Upgrade

To ensure you're using the latest features and fixes, upgrade to the latest `ultralytics` package in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in any of the following verified environments with preinstalled dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Support

Feel free to connect with the Ultralytics community for additional assistance:
- Join our [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time discussions
- Visit [Discourse](https://community.ultralytics.com/) for in-depth conversations
- Check out our [Subreddit](https://reddit.com/r/Ultralytics) for community insights

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If the CI badge above is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests ensure correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response to guide you efficiently. An Ultralytics engineer will follow up with you soon for further assistance. Thank you for your patience and for being part of the Ultralytics community! 😊

**Comment by glenn-jocher** [comment]
Hello @yangershuai627, thanks for reaching out with your question about handling background images in the DOTA dataset.

Including background images (empty samples) during training is generally beneficial as it helps the model learn to reduce false positives. The Ultralytics training pipeline is designed to handle these images.

The script used for splitting DOTA images, as detailed in the [DOTA Dataset with OBB documentation](https://docs.ultralytics.com/datasets/obb/dota-v2/#split-dota-images) and the [`ultralytics.data.split_dota.crop_and_save` reference](https://docs.ultralytics.com/reference/data/split_dota/#ultralytics.data.split_dota.crop_and_save), includes an `allow_background_images` parameter (defaulting to `True`) which controls whether these background patches are saved.

While an extremely high ratio of background images could potentially influence training dynamics, they are typically included. The difference in mAP you observed might stem from other factors like hyperparameters, training duration, or specific model configurations. We generally recommend keeping the background images unless you have specific reasons to exclude them.

**Comment by yangershuai627** [comment]
> Hello [@yangershuai627](https://github.com/yangershuai627), thanks for reaching out with your question about handling background images in the DOTA dataset.
> 
> Including background images (empty samples) during training is generally beneficial as it helps the model learn to reduce false positives. The Ultralytics training pipeline is designed to handle these images.
> 
> The script used for splitting DOTA images, as detailed in the [DOTA Dataset with OBB documentation](https://docs.ultralytics.com/datasets/obb/dota-v2/#split-dota-images) and the [`ultralytics.data.split_dota.crop_and_save` reference](https://docs.ultralytics.com/reference/data/split_dota/#ultralytics.data.split_dota.crop_and_save), includes an `allow_background_images` parameter (defaulting to `True`) which controls whether these background patches are saved.
> 
> While an extremely high ratio of background images could potentially influence training dynamics, they are typically included. The difference in mAP you observed might stem from other factors like hyperparameters, training duration, or specific model configurations. We generally recommend keeping the background images unless you have specific reasons to exclude them.

@glenn-jocher Thank you for your answer. It's very helpful.

**Comment by glenn-jocher** [comment]
You're welcome, @yangershuai627! Glad I could help. Let us know if any other questions come up.
